### PR TITLE
update README and .xml application files

### PR DESCRIPTION
**Pull request by GiuliaP**
With this PR I added a description of the demo in the README and updated the .xml application files.

In particular:
- I replaced the `sparseCoder` module with `caffeCoder`
- I renamed the various .xml files to express their differences
- I added the .xml to run the demo using `SFM` + `dispBlobber` instead of `motionCUT`
- I cleaned up spurious things (e.g. the tags)

One note is that even if I left the .xml that uses the `GurlClassificationBatch` in place of the usual `linearClassifier` module, I couldn't actually find this module in robotology.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +76,8 @@ On The Fly Recognition
 - [Hierarchical Image Representation](https://github.com/robotology/himrep)
 - [speechRecognizer](https://github.com/robotology/speech)
```
Comment (pattacini) [inline README.md:79]: I'd add [`segmentation`](https://github.com/robotology/segmentation) too as external dependency

**Comment by pattacini** [comment]
I guess `GurlClassificationBatch` is hosted in https://github.com/LCSL/GURLS.
Aside from a very minor comment, I believe you can merge this PR.

**Comment by GiuliaP** [comment]
I couldn't find GURLS module, can you point me to it? Added `segmentation` dependency.

**Comment by pattacini** [comment]
Mine was only a guess, I don't know :smile: 
Maybe @Ommac or @cciliber might know ... (another guess)

**Comment by raffaello-camoriano** [comment]
Hi,
As I have already told @GiuliaP, there is no such module in LCSL/GURLS.

**Comment by elen4** [comment]
I think that the GURLS-based modules are still on the old SVN repository on SourceForge (e.g., http://sourceforge.net/p/robotcub/code/HEAD/tree/trunk/iCub/contrib/src/GURLSwrappers/GurlsClassification/batch/ ), I don't know if they were ever ported to git/GitHub

**Comment by GiuliaP** [comment]
Ok! We could merge this PR as it is hence, and then we can test the GURLS module later in the following months (with @Ommac we are planning to port some GURLS functionalities on the robot, and we could start looking at this module). Is it ok for you, @pattacini?

**Comment by pattacini** [comment]
Yes, definitely :+1:

**Pull request merged**